services/discordService.js

```diff
@@ -103,7 +103,40 @@ const setUserDiscordNickname = async (userName, discordId) => {
     };
   } catch (err) {
     logger.error("Error in updating discord Nickname", err);
-    throw err;
+    throw new Error(err);
+  }
+};
+
+/**
+ * Deletes a group role from the Discord server.
+ * This function sends a DELETE request to the Discord API to remove the role.
+ * It's part of the soft delete process, where we remove the role from Discord
+ * but keep a record of it in our database.
+ *
+ * @param {string} roleId - The Discord ID of the role to be deleted
+ * @returns {Promise<Object>} The response from the Discord API
+ * @throws {Error} If the deletion fails or there's a network error
+ */
+
+const deleteGroupRoleFromDiscord = async (roleId) => {
+  try {
+    const authToken = generateAuthTokenForCloudflare();
+    const response = await fetch(`${DISCORD_BASE_URL}/roles/${roleId}?dev=true`, {
```

Comment:
create this url as constant variable and use it here

Reply:
I think this is fine, take reference from other function in the file

Reply:
I think this is fine, take reference from other function in the file

Reply:
No, Since we have to change this url after it is in prod. We have to modify this function again.
In case of variable, this would not be the case

Reply:
Only once right? Wouldn't effect on 1 function, either way we have to remove dev flag in prod.

Reply:
It is better to update a constant variable rather than a working function.

Reply:
Resolved